### PR TITLE
CB-10749 Use cordova-common.CordovaLogger in cordova-android

### PR DESCRIPTION
**Pull request by daserge**
[Jira issue](https://issues.apache.org/jira/browse/CB-10749)

### CODE REVIEW

**Comment by daserge** [comment]
@vladimir-kotikov, could you please take a look once more?

File: bin/templates/cordova/loggingHelper.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+var CordovaLogger = require('cordova-common').CordovaLogger;
+
+module.exports = {
+    adjustLoggerLevel: function (opts) {
+        if (opts instanceof Array) {
+            opts.silent = opts.indexOf('--silent') !== -1;
+            opts.verbose = opts.indexOf('--verbose') !== -1;
```
Comment (vladimir-kotikov) [inline bin/templates/cordova/loggingHelper.js]: We also have `-d` shortcut for `--verbose`

File: bin/templates/cordova/Api.js [hunk 2]
```diff
@@ -29,11 +29,23 @@ var AndroidProject = require('./lib/AndroidProject');
...
 var PluginInfoProvider = require('cordova-common').PluginInfoProvider;
 
-var ConsoleLogger = require('./lib/ConsoleLogger');
 var pluginHandlers = require('./lib/pluginHandlers');
+var CordovaLogger = require('cordova-common').CordovaLogger;
+var selfEvents = require('cordova-common').events;
 
 var PLATFORM = 'android';
 
+function setupEvents(externalEventEmitter) {
```
Comment (vladimir-kotikov) [inline bin/templates/cordova/Api.js]: Could you make this function return the emitter instance? `setupEvents(events); events = events || selfEvents;` -> `events = setupEvents(events);`

File: bin/templates/cordova/Api.js [hunk 3]
```diff
@@ -91,8 +102,11 @@ function Api(platform, platformRootDir, events) {
  *   instance or rejected with CordovaError.
  */
 Api.createPlatform = function (destination, config, options, events) {
+    setupEvents(events);
```
Comment (vladimir-kotikov) [inline bin/templates/cordova/Api.js]: See previous comment.

File: bin/templates/cordova/Api.js [hunk 4]
```diff
@@ -91,8 +102,11 @@ function Api(platform, platformRootDir, events) {
...
+    events = events || selfEvents;
```
Comment (vladimir-kotikov) [inline bin/templates/cordova/Api.js]: See previous comment

**Comment by vladimir-kotikov** [comment]
LGTM apart from few minor comments

@daserge, it'd be nice to have `adjustLoggerLevel` as a method of `CordovaLogger`. Could you please add corresponding change when possible?

File: bin/templates/cordova/loggingHelper.js (hunk 1 above)
Comment (daserge) [inline bin/templates/cordova/loggingHelper.js]: Added, thanks!

File: bin/templates/cordova/Api.js (hunk 2 above)
Comment (daserge) [inline bin/templates/cordova/Api.js]: Updated.

**Comment by vladimir-kotikov** [comment]
LGTM

**Comment by daserge** [comment]
Thanks for the review!
Added https://issues.apache.org/jira/browse/CB-10791 and https://github.com/apache/cordova-lib/pull/405 for the `adjustLoggerLevel` method.

**Pull request merged**